File: xx_docs/selenium/1 - Selenium Testing Tools Cookbook [arquivo-apoio]/Chapter 7/Python/android_test.py
from selenium import webdriver
import time, unittest

class OnAndroid (unittest.TestCase):
	def setUp(self)	:
		self.driver = webdriver.Remote(command_executor='http://localhost:8080/wd/hub', desired_capabilities = webdriver.DesiredCapabilities.ANDROID)
		
	def test_bmi_calc_mobile_version(self):
		driver = self.driver
		driver.get("http://dl.dropbox.com/u/55228056/mobilebmicalculator.html")
		
		heightField = driver.find_element_by_name("heightCMS")	
		heightField.send_keys("181")
		
		weightField = driver.find_element_by_name("weightKg")	
		weightField.send_keys("80")
		
		calculateButton = driver.find_element_by_id("Calculate")	
		calculateButton.click()
		
		bmiLabel = driver.find_element_by_name("bmi")
		bmiCategoryLabel = driver.find_element_by_name("bmi_category")
		
		self.assertEqual("24.4",bmiLabel.get_attribute("value"))
		self.assertEqual("Normal",bmiCategoryLabel.get_attribute("value"))

	def tearDown(self):
		self.driver.quit()
		
if __name__ == "__main__":
	unittest.main()

File: xx_docs/selenium/1 - Selenium Testing Tools Cookbook [arquivo-apoio]/Chapter 7/Ruby/android_test.rb
require 'rubygems'
require 'selenium-webdriver'
require 'test/unit'

class AndroidTest < Test::Unit::TestCase
	def setup
		@driver = Selenium::WebDriver.for :android
		@driver.get 'http://dl.dropbox.com/u/55228056/mobilebmicalculator.html'
		@verification_errors = []
	end
	
	def test_bmi_calc_mobile_version
		height = @driver.find_element :name => "heightCMS"
		height.send_keys '181'

		weight = @driver.find_element :name => "weightKg"
		weight.send_keys  '80'

		calculateButton = @driver.find_element :id =>"Calculate"
		calculateButton.click

		bmi = @driver.find_element :name =>"bmi"
		bmi_category = @driver.find_element :name =>"bmi_category"
		
		assert_equal '24.4', bmi.attribute('value')
		assert_equal 'Normal', bmi_category.attribute('value')
	end
	
	def teardown
		@driver.quit
		assert_equal [], @verification_errors
	end
end